### ISSUE TITLE
AutomationEngine.onScheduleConditionsChanged NPE

### ISSUE THREAD

**Issue by xbao** [urbanairship/android-library#74]
Hello, I'm seeing some crashes come in from the UrbanAirship SDK. 

Lib version: 9.0.2
Android versions: seen on 4.4-8
Frequency: ~1% of users

Stacktrace:
```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.os.Handler.post(java.lang.Runnable)' on a null object reference
       at com.urbanairship.automation.AutomationEngine.onScheduleConditionsChanged(AutomationEngine.java:868)
       at com.urbanairship.iam.InAppMessageManager$4.com.urbanairship.automation.AutomationEngine.checkPendingSchedules(InAppMessageManager.java:1510)
       at com.urbanairship.ActivityMonitor$1.onActivityResumed(ActivityMonitor.java:70)
       at android.app.Application.dispatchActivityResumed(Application.java:216)
       at android.app.Activity.onResume(Activity.java:1306)
       at android.support.v4.app.FragmentActivity.onResume(FragmentActivity.java:446)
       at nz.co.skytv.sporthighlights.BaseActivity.onResume(BaseActivity.java:65)
       at nz.co.skytv.sporthighlights.loading.LoadingActivity.onResume(LoadingActivity.java:84)
       at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1279)
       at android.app.Activity.performResume(Activity.java:7017)
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3611)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3676)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2876)
       at android.app.ActivityThread.-wrap12(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1567)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:156)
       at android.app.ActivityThread.main(ActivityThread.java:6577)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:941)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:831)
```

Happy to answer any questions

**Comment by rlepinski** [comment]
@xbao This issue was fixed in 9.0.4, but please update to 9.0.6.

**Comment by xbao** [comment]
Will do, thanks!
